source/math/FloatMatrix.ooc

```diff
@@ -45,6 +45,36 @@ FloatMatrix : cover {
 		result
 	}
 
+	setColumn: func (x: Int, values: ...) {
+		if (x < 0 || x >= this width)
+			raise("Column index out of range in FloatMatrix setColumn")
+		if (values count != this height)
+			raise("Number of values must match matrix height in FloatMatrix setColumn")
+		y := 0
+		values each (|value|
+			match value {
+				case floatValue: Float => this[x, y] = floatValue
+				case => raise("Values must be of type Float in FloatMatrix setColumn")
+			}
+			y += 1
+		)
+	}
+
+	setColumn: func ~floatpoint3d (x: Int, vector: FloatPoint3D) {
+		if (this height != 3)
+			raise("Matrix must be of height 3 in FloatMatrix setColumn")
```

Comment:
That seems like an unnecessary limitation... Can't we add a `y` parameter, bounds check inside a `safe` block and allow the caller to write 3 values horizontally anywhere in a matrix?